serialization/fontDataset.py

```diff
@@ -0,0 +1,89 @@
+from os.path import join 
+import numpy as np
+import pandas as pd
+class Dataset(data.Dataset): 
+
+    def __init__(self, proto_dir, glyph_list, dimensions): 
+        '''
+        Parameters
+        ----------
+
+        proto_dir : directory of where all the glyph protos are located
+
+        glyph_list : file path that contains list of glyphs
+        '''
+        self.proto_dir = proto_dir
+        self.glyph_list = pd.read_csv(glyph_list,header=None)
+        self.dimensions = dimensions
+    def __len__(self):
+        return len(self.glyph_list)
+
+    def __getitem__(self,index): 
+        glyph = self.glyph_list.iloc[[index]]
+        bezier_points, contour_locations = readSerializedProto(glyph,self.proto_dir)
+        
+        # we can call a little function here to reshape and also add padding
+        # or alternativly it can be called on the other end
+        return (rollPadReshape(bezier_points,
+            contour_locations,self.dimensions), str(glyph))
+
+
+
+
+
+def rollPadReshape(bezier_points,contour_locations, dimensions):
+        '''
+        Parameters
+        ----------
+
+        bezier_points : the flattned bezier points returned by the proto
+        contour_locations: the number of curves in each of the contours
+        dimensions: the output dimensions we want to come out, The assumption is
+        that this follows (contours,curves,points=3,coordinates=2)
+        Returns:
+        -------
+        An numpy array with the dimensions specefied.
+        '''
+        
+        # in order to save computation we are going to try to do this all with
+        # one for loop
+        
+        reshaped_glyph = np.full(dimensions,999)
+        start = 0
+        num_contours = min(len(contour_locations), dimensions[0])
+        for i in range(num_contours):
+            contour = contour_locations[i]
+            roll_num = np.random.randint(20)
+            # isolate the points for this contour
+            end = contour * 3 * 2 + start # conversion from curves to points
+            current_points = bezier_points[start:end]
+            # fold it into the correct shape and pop it in.
+            folded_values = np.reshape(current_points,(1,contour,3,2))
+            curves_dim = min(contour, dimensions[1])
+            reshaped_glyph[i,:curves_dim,:,:] = folded_values[0,:curves_dim,:,:]
+            start = end
+        return reshaped_glyph
+
+
+def readSerializedProto(glyph, proto_dir):
+    '''
+     This function takes a glyph row from the glyph list csv and then returns
+     the corresponding bezier information to be later reshaped
+
+     Paramaters
+     ---------
+     glyph: a pandas object, a specefic row of the glyph list
+     proto_dir: the directory with the corresponding glyph protos
+    '''
+
+    index = glyph[1].values[0]
+    glyph = glyph[0].values[0]
+    glyph_proto = eval('font_pb2.{}()'.format(glyph))
+    glyph_proto.ParseFromString(open(join(proto_dir,glyph),'rb').read())
```

Comment:
You don't want to read the proto in this func b/c the whole proto will be read every time you need to get a batch. Read the proto in the dataloader init

Reply:
Actually - let me back up a bit. Reading here is fine with a sufficiently sharded dataset.

Reply:
Okay, should I move it to the init? 

each of the proto files are about 121mb